src/bitcask.erl

```diff
@@ -1930,10 +1969,18 @@ expiry_merge([File | Files], LiveKeyDir, KT, Acc0) ->
     FileId = bitcask_fileops:file_tstamp(File),
     Fun = fun({tombstone, _}, _, _, Acc) ->
                   Acc;
-             (K, Tstamp, {Offset, _TotalSz}, Acc) ->
-                  bitcask_nifs:keydir_remove(LiveKeyDir, KT(K), Tstamp, FileId,
-                                             Offset),
-                  Acc
+             (K0, Tstamp, {Offset, _TotalSz}, Acc) ->
+                  K = try KT(K0) catch TxErr -> {key_tx_error, TxErr} end,
+                  case K of
+                      {key_tx_error, KeyTxErr} ->
+                          error_logger:error_msg("Invalid key on merge ~p: ~p",
+                                                 [K0, KeyTxErr]),
+                          Acc;
+                      _ ->
+                          bitcask_nifs:keydir_remove(LiveKeyDir, K, Tstamp,
+                                                     FileId, Offset),
+                          Acc
+                  end
         end,
     case bitcask_fileops:fold_keys(File, Fun, ok, default) of
```

Comment:
I'm not sure what you mean. Acc is an input accumulator. It is not assigned in the case statement. It is returned untouched just like the original code did.